src/v/security/audit/audit_log_manager.cc

```diff
@@ -504,7 +517,9 @@ ss::future<> audit_client::produce(
       records.size(),
       total_size);
 
-    auto reserved = co_await ss::get_units(_send_sem, total_size);
+    auto timepoint = timeout.has_value() ? ss::timer<>::clock::now() + *timeout
+                                         : ss::timer<>::time_point::max();
+    auto reserved = co_await ss::get_units(_send_sem, total_size, timepoint);
```

Comment:
nit: if you don't want to set a timeout, then using the non-timeout interface would be better since you won't actually insert anything into the timer wheel.